io/shared/src/test/scala/fs2/io/file/PathSuite.scala

```diff
@@ -38,12 +38,31 @@ class PathSuite extends Fs2IoSuite {
     root <- Gen.oneOf("/", "")
   } yield names.foldLeft(Path(root))((p, n) => p / Path(n)))
 
+  val nonEmptyPath = arbitraryPath.arbitrary.filter(
+    _.toString.nonEmpty
+  )
+
   implicit val cogenPath: Cogen[Path] =
     Cogen.cogenList[String].contramap(_.names.map(_.toString).toList)
 
   test("construction") {
     assertEquals(Path("foo/bar"), Path("foo") / "bar")
     assertEquals(Path("/foo/bar"), Path("/foo") / "bar")
+
+    val sepLengthGen = Gen.choose(0, 1024)
+
+    forAll(sepLengthGen)((sepLength: Int) =>
+      assertEquals(
+        Path("/".repeat(sepLength)).toString,
+        "/"
+      )
+    )
```

Comment:
Wait a minute, is this test passing even when `sepLength = 0`?

Reply:
No, it should not. After investigation I found out that scalacheck is mishandled in this suite. Tried to fix that